data-raw/northern-ireland-hsct-secondary-care.R

```diff
@@ -0,0 +1,145 @@
+# Northern Ireland - Secondary Care Indicators
+# Health and Social Care Trusts
+
+# ---- Load libs & helpers ----
+library(tidyverse)
+library(lubridate)
+library(sf)
+library(healthyr)
+library(compositr)
+library(ggridges)
+library(devtools)
+library(readODS)
+library(geographr)
+
+hsct <- boundaries_trusts_ni18 |>
+  st_drop_geometry()
+
+# Population counts
+# Source: https://www.ninis2.nisra.gov.uk/public/AreaProfileReportViewer.aspx?FromAPAddressMulipleRecords=Northern@@Northern@20?
+
+url <- "https://www.ninis2.nisra.gov.uk/Download/Population/Population%20Estimates%20Broad%20Age%20Bands%20(administrative%20geographies).ods"
+
+pop_raw <- download_file(url, ".ods")
+
+pop_ni_hsct <- read_ods(
+  pop_raw,
+  sheet = "HSCT",
+  range = "A4:C9"
+) |>
+  select(
+    trust18_name = "HSCT",
+    trust18_code = "HSCT Code",
+    population = "Persons"
+  )
+
+# ---- RTT ----
+# Higher = worse performance
+rtt <- ni_rtt_hsct |>
+  ungroup() |> # was grouped in ni_rtt_hsct
+  select(
+    trust18_name = "hsct22_name",
+    date,
+    waits_over_18_weeks
+  ) |>
+  filter(date >= max(date) %m-% months(3)) |> # Last 3 months
+  left_join(pop_ni_hsct) |>
+  mutate(percent = waits_over_18_weeks / population) |>
+  group_by(trust18_name) |>
+  summarise(
+    number = sum(waits_over_18_weeks),
+    percent = sum(percent, na.rm = FALSE)
+  ) |>
+  mutate(
+    variable = "Referral to treatment \nwaiting times (Mar 22 - Jun 22)",
+    .after = trust18_name
+  ) |>
+  filter(!trust18_name %in% c("DPC", "Day Case Procedure Centre"))
```

Comment:
While it doesn't matter in this instance, better practice would have been to move this filter call to the beginning of this code block. This means the computation time would have been reduced for subsequent calls to mutate and summarise (as the data frame it would be operating on would be smaller). You can leave this here for now though.

Reply:
Yes, that is understandable!